### PR TITLE
[Backport Foxy] Update QD

### PR DESCRIPTION
**Pull request by ahcorde**

### CODE REVIEW

File: rmw_fastrtps_cpp/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -104,31 +119,31 @@ There is no API testing in `rmw_fastrtps_cpp`.
...
 
 It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
 It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
 
 ### Direct Runtime Non-ROS Dependencies [5.iii]
 
 `rmw_fastrtps_cpp` has the following runtime non-ROS dependencies.
-* `fastcdr`
-* `fastrtps`
+* `fastcdr`: *eProsima Fast CDR* claims to be Quality Level 2. For more information, please refer to its [QUALITY DECLARATION](https://github.com/eProsima/Fast-CDR/blob/foxy/QUALITY.md)
```
Comment (JLBuenoLopez-eProsima) [inline rmw_fastrtps_cpp/QUALITY_DECLARATION.md:145]:
```suggestion
* `fastcdr`: *eProsima Fast CDR* claims to be Quality Level 2. For more information, please refer to its [QUALITY DECLARATION](https://github.com/eProsima/Fast-CDR/blob/master/QUALITY.md)
```
Foxy ros2.repos uses release tag v1.0.13 for eProsima Fast CDR. Therefore, you can link to the master Quality Declaration.

File: rmw_fastrtps_cpp/QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -104,31 +119,31 @@ There is no API testing in `rmw_fastrtps_cpp`.
...
+* `fastrtps`: *eProsima Fast DDS* claims to be Quality Level 2. For more information, please refer to its [QUALITY DECLARATION](https://github.com/eProsima/Fast-DDS/blob/foxy/QUALITY.md)
```
Comment (JLBuenoLopez-eProsima) [inline rmw_fastrtps_cpp/QUALITY_DECLARATION.md:146]:
```suggestion
* `fastrtps`: *eProsima Fast DDS* claims to be Quality Level 2. For more information, please refer to its [QUALITY DECLARATION](https://github.com/eProsima/Fast-DDS/blob/2.0.x/QUALITY.md)
```

Foxy ros2.repos uses branch <2.0.x> for eProsima Fast DDS.

File: rmw_fastrtps_cpp/README.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
 
-`rmw_fastrtps_cpp` implements the ROS middleware interface using eProsima FastRTPS static code generation in C++.
+`rmw_fastrtps_cpp` implements the ROS middleware interface using eProsima Fast DDS static code generation in C++.
 
 For more information see the repository level [README](../README.md)
 
 ## Quality Declaration
 
-This package claims to be in the **Quality Level 4** category, see the [Quality Declaration](QUALITY_DECLARATION.md) for more details.
+This package claims to be in the **Quality Level 2** category, see the [Quality Declaration](QUALITY_DECLARATION.md) for more details.
```
Comment (JLBuenoLopez-eProsima) [inline rmw_fastrtps_cpp/README.md:9]: There is an inconsistency between this claim (QL2) and the one in the Quality Declaration for `rmw_fastrtps_cpp` package (QL3)

File: rmw_fastrtps_dynamic_cpp/QUALITY_DECLARATION.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+The Quality Declarations for the last two dependencies are not available currently.
+
+It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
+It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+
+### Direct Runtime Non-ROS Dependencies [5.iii]
+
+`rmw_fastrtps_cpp` has the following runtime non-ROS dependencies.
+* `fastcdr`: [QUALITY DECLARATION](https://github.com/eProsima/Fast-CDR/blob/foxy/QUALITY.md)
+* `fastrtps`: [QUALITY DECLARATION](https://github.com/eProsima/Fast-DDS/blob/foxy/QUALITY.md)
```
Comment (JLBuenoLopez-eProsima) [inline rmw_fastrtps_dynamic_cpp/QUALITY_DECLARATION.md:143]: See comments above about these two links

File: rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md [hunk 5]
```diff
@@ -124,8 +141,8 @@ It also has several test dependencies, which do not affect the resulting quality
 ### Direct Runtime Non-ROS Dependencies [5.iii]
 
 `rmw_fastrtps_shared_cpp` has the following runtime non-ROS dependencies.
-* `fastcdr`
-* `fastrtps`
+* `fastcdr`: [QUALITY DECLARATION](https://github.com/eProsima/Fast-CDR/blob/foxy/QUALITY.md)
+* `fastrtps`: [QUALITY DECLARATION](https://github.com/eProsima/Fast-DDS/blob/foxy/QUALITY.md)
```
Comment (JLBuenoLopez-eProsima) [inline rmw_fastrtps_shared_cpp/QUALITY_DECLARATION.md:145]: See comments above about these two links.

**Review by brawner — APPROVED** [review]

**Pull request merged**